### PR TITLE
feat: add Service and Upstream refactoring

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?

add Service and Upstream refactoring

### CODE REVIEW

File: api/route/base.go [hunk 1]
```diff
@@ -44,16 +46,18 @@ func SetUpRouter() *gin.Engine {
 	AppendHealthCheck(r)
 	AppendAuthentication(r)
 	//AppendRoute(r)
-	AppendSsl(r)
+	//AppendSsl(r)
 	AppendPlugin(r)
-	AppendUpstream(r)
+	//AppendUpstream(r)
```
Comment (juzhiyuan) [inline api/route/base.go:51]: Why not remove them directly?

**Review by juzhiyuan — APPROVED** [review]
Looks good to me from the codes style, but needs @ShiningRush 's help to review the logics.

File: api/route/base.go (hunk 1 above)
Comment (nic-chen) [reply api/route/base.go:51]: we need them to debug or compare.will remove them once when complete the refactoring work.

**Review by moonming — APPROVED** [review]

**Review by ShiningRush — CHANGES_REQUESTED** [review]
Pending comment again, my fault : (

**Comment by ShiningRush** [comment]
@nic-chen The pr has merged, we can open a new pr to fix problems.

**Comment by juzhiyuan** [comment]
cc @nic-chen

**Pull request merged**